app/liquid/liquid_renderer.rb

```diff
@@ -58,22 +35,44 @@ def personalization_data
     }.deep_stringify_keys
   end
 
-  # TODO
-  # move to the member's model?
-  def stored_payment_methods
-    return [] unless @member && @member.customer
-    @member.customer.payment_methods.stored.map do |m|
-      {
-        id: m.id,
-        last_4: m.last_4,
-        instrument_type: m.instrument_type,
-        card_type: m.card_type,
-        email: m.email
-      }
+  private
+
+  def render_layout(layout)
+    cache = Cache.new(@page.cache_key, layout.try(:cache_key))
+    cache.fetch do
+      Liquid::Template.parse(layout.content).render(markup_data).html_safe
     end
   end
 
-  private
+  # this is all of the data that is needed to render the
+  # liquid page. the only parts that change on each request
+  # are not used when rendering markup
+  def markup_data
+    {
+      images:        images,
+      named_images:  named_images,
+      primary_image: image_urls(@page.image_to_display),
+      shares:        Shares.get_all(@page),
+      follow_up_url: follow_up_url
+    }
+    .merge(@page.liquid_data)
+    .merge(LiquidHelper.globals(page: @page))
+    .merge(plugin_data)
+    .deep_stringify_keys
+  end
+
+  def images
+    @page.images.map { |img| image_urls(img) }
+  end
+
+  def named_images
+    named = {}
+    @page.images.each do |img|
```

Comment:
absolutely, but I touch the implementation of this method, I just moved it to make it private ^_^